geode-core/src/main/java/org/apache/geode/management/internal/beans/stats/VMStatsMonitor.java

```diff
@@ -103,22 +92,31 @@ private void refreshStats() {
         lastProcessCpuTime = cpuTime;
         return;
       }
-      long systemTime = System.currentTimeMillis();
-
-      // 10000 = (Nano conversion factor / 100 for percentage)
-      long denom = (systemTime - lastSystemTime) * 10000;
-
-      float processCpuUsage = (float) (cpuTime - lastProcessCpuTime) / denom;
 
+      long systemTime = currentTimeMillis();
       lastSystemTime = systemTime;
       lastProcessCpuTime = cpuTime;
-      cpuUsage = processCpuUsage;
+      cpuUsageBits.set(Float.floatToIntBits(calculateCpuUsage(systemTime, cpuTime)));
     }
-
   }
 
-  public float getCpuUsage() {
-    return cpuUsage;
-  }
+  @Override
+  public void handleNotification(StatisticsNotification notification) {
+    for (StatisticId statId : notification) {
+      StatisticDescriptor descriptor = statId.getStatisticDescriptor();
+      String name = descriptor.getName();
+      Number value;
+
+      try {
+        value = notification.getValue(statId);
+      } catch (StatisticNotFoundException e) {
+        value = 0;
+      }
 
+      log(name, value);
+      statsMap.put(name, value);
```

Comment:
It looks like `statsMap` is also racy -- it should probably be a `ConcurrentHashMap<String,AtomicLong>` or so. And we should probably get rid of `DefaultHashMap`. Though it looks like you've picked up that ticket, GEODE-5523, as well.